instrumentation/opentelemetry-instrumentation-elasticsearch/src/opentelemetry/instrumentation/elasticsearch/__init__.py

```diff
@@ -103,6 +103,7 @@ def response_hook(span, response):
 es_transport_split = elasticsearch.VERSION[0] > 7
```

Comment:
I know `es_transport_split` is not introduced here but we are adding more checks that use it in this PR. Could we have an added comment that briefly summarizes how Otel instrumentation is affected by changes in elasticsearch 8?

Reply:
I've added 3 more users of es_transport_split:
- 1 in uninstrument to match what was already done in instrument
- 1 in the request hook to normalize args that are not json serializable
- 1 is for taking into account the wrapped `perform_request` have a different return value
- 1 is for settings the status error in case of failure since it has been moved to upper layers, I already added a comment here

From a user point of view I think the request hook parameters are the most impacting change, added a comment.

Reply:
BTW this is testing against 8.12 that has not the native OTel instrumentation. With an 8.13 client with the native instrumentation active the span created would be different. Hopefully we'll converge when updating ours to more recent semantic conventions.